### PR TITLE
remove tables negative side handling

### PR DESCRIPTION
**Pull request by tharukshi1bt**

### CODE REVIEW

File: pptx_generator/src/handlers/generate_pptx.py [hunk 1]
```diff
@@ -117,68 +117,69 @@ def generate_pptx(event, context):
...
-                "statusCode": 200,
-                "body": json.dumps(body)
-            }
-
-            return response
-        else:
-            response = {
-                "statusCode": 500,
-                "message":""
+                except ClientError as e:
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: Check the specific exception type and then send the error response according to that. (ex: if the exception type is "bucket not found" then send "destination bucket cannot be found" response message).

Handle every possible exceptions in every places in the handler function.

File: pptx_generator/src/handlers/generate_pptx.py [hunk 2]
```diff
@@ -117,68 +117,69 @@ def generate_pptx(event, context):
...
+                       "statusCode":404,
+                       "message":"Destination bucket cannot be found"
+                    }
+                    return response
+                    
+                
                 
+        except ClientError as e:
+            logging.error(e)
+            return False
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: Handle error response message properly.

File: pptx_generator/src/handlers/generate_pptx.py [hunk 3]
```diff
@@ -117,68 +117,69 @@ def generate_pptx(event, context):
...
+            "statusCode": 200,
+            "body": json.dumps(body)
+        }
 
-            return response
+        return response
+    
+    except ClientError as e:
+         logging.error(e)
+         response= {
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: Handle error response message based on the exception type..

**Review by thivankanimesh1bt — CHANGES_REQUESTED** [review]

File: pptx_generator/src/handlers/generate_pptx.py (hunk 1 above)
Comment (thivankanimesh1bt) [reply pptx_generator/src/handlers/generate_pptx.py]: pending...

File: pptx_generator/src/handlers/generate_pptx.py (hunk 2 above)
Comment (thivankanimesh1bt) [reply pptx_generator/src/handlers/generate_pptx.py]: pending...

File: pptx_generator/src/handlers/generate_pptx.py (hunk 3 above)
Comment (thivankanimesh1bt) [reply pptx_generator/src/handlers/generate_pptx.py]: pending...

**Review by thivankanimesh1bt — CHANGES_REQUESTED** [review]

**Review by thivankanimesh1bt — APPROVED** [review]

**Review by GayashanSameera — APPROVED** [review]

File: pptx_generator/src/handlers/generate_pptx.py [hunk 4]
```diff
@@ -119,13 +122,34 @@ def generate_pptx(event, context):
...
+
+        # f = open('/tmp/{template_key}.pptx', "wb")
+        # f.write(data)
+        # f.close()
+        
+        response = s3_client.get_object(Bucket=POC_PPTX_BUCKET, Key='demo1.pptx')
         data = response['Body'].read()
 
-        f = open('/tmp/{template_key}.pptx', "wb")
+        f = open('/tmp/demo1.pptx', "wb")
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: Why demo1.pptx is hardcoded?

File: pptx_generator/src/handlers/generate_pptx.py [hunk 5]
```diff
@@ -119,13 +122,34 @@ def generate_pptx(event, context):
...
     try:
-        response = s3_client.get_object(Bucket=POC_PPTX_BUCKET, Key=str(f"{template_key}.pptx"))
+        # response = s3_client.get_object(Bucket=POC_PPTX_BUCKET, Key=str(f"{template_key}.pptx"))
+        # data = response['Body'].read()
+
+        # f = open('/tmp/{template_key}.pptx', "wb")
+        # f.write(data)
+        # f.close()
+        
+        response = s3_client.get_object(Bucket=POC_PPTX_BUCKET, Key='demo1.pptx')
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: why demo1.pptx is hardcoded?

File: pptx_generator/src/handlers/generate_pptx.py [hunk 6]
```diff
@@ -119,13 +122,34 @@ def generate_pptx(event, context):
...
+        print(res)
+        cwd = os.getcwd()
+        print("Current working directory: {0}".format(cwd))
+        # os.chdir('/tmp')
+        # cwd1 = os.getcwd()
+        # print("Current working directory: {0}".format(cwd1))
+        subprocess.call('dir && ppt2pdf file demo1.pptx', shell=True, cwd='/tmp')
+        subprocess.call('ppt2pdf file demo1.pptx', shell=True)
+        
+        presentation_object = Presentation('/tmp/demo1.pptx')
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: Dont hardcode demo.pptx

File: pptx_generator/src/handlers/generate_pptx.py [hunk 7]
```diff
@@ -119,13 +122,34 @@ def generate_pptx(event, context):
...
+        for path in os.listdir('/tmp'):
+         if os.path.isfile(os.path.join('/tmp', path)):
+            res.append(path)
+        print(res)
+        cwd = os.getcwd()
+        print("Current working directory: {0}".format(cwd))
+        # os.chdir('/tmp')
+        # cwd1 = os.getcwd()
+        # print("Current working directory: {0}".format(cwd1))
+        subprocess.call('dir && ppt2pdf file demo1.pptx', shell=True, cwd='/tmp')
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: dont hardcode demo1.pptx

**Review by thivankanimesh1bt — CHANGES_REQUESTED** [review]

File: pptx_generator/src/handlers/generate_pptx.py [hunk 8]
```diff
@@ -119,13 +122,34 @@ def generate_pptx(event, context):
...
 
-        f = open('/tmp/{template_key}.pptx', "wb")
+        f = open('/tmp/demo1.pptx', "wb")
         f.write(data)
         f.close()
-        presentation_object = Presentation('/tmp/{template_key}.pptx')
+        res = []
+        for path in os.listdir('/tmp'):
+         if os.path.isfile(os.path.join('/tmp', path)):
+            res.append(path)
```
Comment (thivankanimesh1bt) [inline pptx_generator/src/handlers/generate_pptx.py]: What is the purpose of this 'res' list and the if condition?

**Review by thivankanimesh1bt — CHANGES_REQUESTED** [review]

**Review by thivankanimesh1bt — APPROVED** [review]

**Pull request merged**